docs/bokeh/source/docs/dev_guide/setup.rst

```diff
@@ -640,6 +640,9 @@ If you keep getting errors after updating an older environment, use
 and reinstall your development environment, following the steps in this guide
 from :ref:`the beginning <contributor_guide_setup_preliminaries>`.
 
+For users on Windows systems, sometimes you may get a "File contains carriage returns at end of line: <file path>" error while trying to push your local branch to your remote branch on Github. If that happens, go to your *source checkout* directory and run the following command:
+ ``git config --global core.autocrlf false``, then delete and re-clone your forked repository. This command preserves the original LF only newlines in all the files in the repository.
```

Comment:
This makes the sentence easier to read, I think:
```suggestion
 ``git config --global core.autocrlf false``, then delete and re-clone your forked repository. This command preserves the original LF-only newlines in all the files in the repository.
```

Reply:
It might be sufficient to just run `git checkout -- .`  at the top level, rather than deleting and re-cloning. But it would be good to verify that is the case before changing, of course.

Reply:
github seems to recommend `git add --renormalize .` (https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings) - seems like there are several ways to do it. I feel like deleting and re-cloning after setting the global git config might be the easiest and "cleanest" way. We could make that the recommended way and then link to the Github docs (and possibly other sources) for more information and alternative solutions.

Reply:
@bryevdv we could also consider adding a .gitattributes file to our repo to configure this repo-wide (https://git-scm.com/docs/gitattributes#_text). This is also mentioned in GitHub's documentation: https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings#per-repository-settings

Reply:
If we can solve this once and for all with a change to `.gitattributes` that would be the best solution for sure. Can you open an issue around this?